lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -1077,9 +1074,11 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
           (innerTy->isObjCExistentialType() && !isCFTypeRef(innerTy))) {
         os << ", weak";
       }
-    } else if (ty->is<UnownedStorageType>()) {
-      os << ", assign";
-    } else if (ty->is<UnmanagedStorageType>()) {
+    } else if (ty->is<UnownedStorageType>()|| ty->is<UnmanagedStorageType>()) {
+      // We treat "unowned" as "unsafe_unretained" (even though it's more like
+      // "safe_unretained") because we want people to think twice about
+      // allowing that object to disappear. "unowned(unsafe)" (and Unmanaged,
+      // handled below) really are "unsafe_unretained".
```

Comment:
Can this reasonably be turned into an exhaustive switch over reference storage kinds?